orttraining/orttraining/test/training_ops/cuda/softmax_test.cc

```diff
@@ -39,79 +40,132 @@ static void TestSoftmax(const std::vector<int64_t>& X_dims,
 TEST(CudaKernelTest, Softmax_SmallTensor_LastAxis) {
   std::vector<int64_t> X_dims{4, 2, 128};
   std::vector<int64_t> Y_dims{4, 2, 128};
-  TestSoftmax(X_dims, Y_dims, 2, false);
+  TestSoftmax<float>(X_dims, Y_dims, 2, false);
 }
 
 TEST(CudaKernelTest, Softmax_SmallTensor_AllAxis) {
   std::vector<int64_t> X_dims{4, 2, 128};
   std::vector<int64_t> Y_dims{4, 2, 128};
-  TestSoftmax(X_dims, Y_dims, 0, false);
-  TestSoftmax(X_dims, Y_dims, 1, false);
+  TestSoftmax<float>(X_dims, Y_dims, 0, false);
+  TestSoftmax<float>(X_dims, Y_dims, 1, false);
 }
 
 // large tensor to check cuda DNN softmax forward
 TEST(CudaKernelTest, Softmax_LargeTensor_LastAxis) {
   std::vector<int64_t> X_dims{8, 16, 2048};
   std::vector<int64_t> Y_dims{8, 16, 2048};
-  TestSoftmax(X_dims, Y_dims, 2, false);
+  TestSoftmax<float>(X_dims, Y_dims, 2, false);
+}
+
+TEST(CudaKernelTest, Softmax_LargeTensor_LastAxis_Float16) {
+  std::vector<int64_t> X_dims{8, 16, 2048};
+  std::vector<int64_t> Y_dims{8, 16, 2048};
+  TestSoftmax<MLFloat16>(X_dims, Y_dims, 2, false, 1e-3, 1e-3);
+}
+
+TEST(CudaKernelTest, Softmax_LargeTensor_LastAxis_Float16_no_power_of_2) {
+  std::vector<int64_t> X_dims{8, 16, 1500};
+  std::vector<int64_t> Y_dims{8, 16, 1500};
+  TestSoftmax<MLFloat16>(X_dims, Y_dims, 2, false, 1e-3, 1e-3);
 }
 
 TEST(CudaKernelTest, Softmax_LargeTensor_AllAxis) {
   std::vector<int64_t> X_dims{8, 16, 512};
   std::vector<int64_t> Y_dims{8, 16, 512};
-  TestSoftmax(X_dims, Y_dims, 0, false);
-  TestSoftmax(X_dims, Y_dims, 1, false);
+  TestSoftmax<float>(X_dims, Y_dims, 0, false);
+  TestSoftmax<float>(X_dims, Y_dims, 1, false);
+}
+
+TEST(CudaKernelTest, Softmax_LargeTensor_AllAxis_Float16) {
+  std::vector<int64_t> X_dims{8, 16, 512};
+  std::vector<int64_t> Y_dims{8, 16, 512};
+  TestSoftmax<MLFloat16>(X_dims, Y_dims, 0, false, 1e-3, 1e-3);
+  TestSoftmax<MLFloat16>(X_dims, Y_dims, 1, false, 1e-3, 1e-3);
+}
+
+TEST(CudaKernelTest, Softmax_LargeTensor_AllAxis_Float16_no_power_of_2) {
```

Comment:
can we follow existing naming format? -->Softmax_LargeTensor_AllAxis_Float16_NoPowerOfTwo? 
check others also.